capa/rules/__init__.py

```diff
@@ -438,7 +485,7 @@ def pop_statement_description_entry(d):
     return description["description"]
 
 
-def build_statements(d, scope: str):
+def build_statements(d, scope: Union[str, Flavor]):
```

Comment:
also here: i think this function should accept `Scope` not `Flavor/Scopes`. build the statements in the context of a single scope. otherwise, the return value should be much more complex: a mapping from flavor to built statements.

Reply:
I agree that using `Scope` instead of `Scopes` is better, but in this case it would get rid of the ability to mix static and dynamic features together; so mixed rules (that specify both a static and dynamic scope) would need to be made up of only features/subscopes common across the static and dynamic flavors.

Reply:
> get rid of the ability to mix static and dynamic features together

would you explain this in a bit more detail?

Reply:
sorry for not being clear. I meant the feature mixing at a scope/subscope level.
correct me if I am wrong, but if the `build_statements()` function knows only of one scope (i.e. it takes in an `str` and not a `Scopes` object), then rules like the one mentioned above would raise an `InvalidRule` exception.

this would be because `build_statements()` would have either function or thread as a scope, so if we set the scope for `build_statements()` to _thread_, then we'd get an `InvalidRule("instruction subscope supported only for function and basic block scope") since instruction is valid for only the function scope.

this would be more pronounced if a rule has static-only features and dynamic-only features mixed at the same level/depth, since the `build_statements()` function wouldn't be able to construct neither a static statement (because of the present dynamic features), and the same for a dynamic statement (because of the present static features).

however, if we pass a `Scopes` object to `build_statements()`, then constructing such rule would be possible: instruction would be valid because it would pass for the parent function scope, and `api` would pass because it's valid for both thread and process scopes. So when the rule gets matched in a dynamic context it'd match only for the features valid for the thread scopes (api).

not sure if my explanation makes sense, but if it isn't then please let me know! looking forward to hear your thoughts on this.